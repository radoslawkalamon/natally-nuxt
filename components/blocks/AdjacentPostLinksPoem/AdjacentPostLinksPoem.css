.adjacent-post-links-poem {
  display: flex;
  gap: var(--gap);
  justify-content: space-between;
}

.adjacent-post-links-poem__previous {
  margin-right: auto;
}

.adjacent-post-links-poem__next {
  margin-left: auto;
}
