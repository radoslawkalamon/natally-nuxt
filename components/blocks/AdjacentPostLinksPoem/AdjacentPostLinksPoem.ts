import Vue from 'vue'
import { DTOMetaPostPoem, DTOMetaPostPoemConstructor } from '@/utils/dto.meta.post.poem'

export default Vue.extend({
  name: 'BlocksAdjacentPostLinksPoem',
  components: {
    ComponentsAdjacentPostLinkNext: () => import(/* webpackChunkName: "components" */'@/components/components/AdjacentPostLinkNext/AdjacentPostLinkNext.vue'),
    ComponentsAdjacentPostLinkPrevious: () => import(/* webpackChunkName: "components" */'@/components/components/AdjacentPostLinkPrevious/AdjacentPostLinkPrevious.vue')
  },
  props: {
    slug: {
      type: String,
      required: true
    }
  },
  data (): Record<string, DTOMetaPostPoemConstructor[]> {
    return {
      articles: []
    }
  },
  async fetch () {
    const articles = await this.$content('poezja-314')
      .sortBy('createdAt', 'asc')
      .surround(this.slug)
      .fetch<DTOMetaPostPoemConstructor>()

    this.articles = articles as DTOMetaPostPoemConstructor[]
  },
  computed: {
    previousPoem (): DTOMetaPostPoem | null {
      return this.articles[0] ? new DTOMetaPostPoem(this.articles[0]) : null
    },
    nextPoem (): DTOMetaPostPoem | null {
      return this.articles[1] ? new DTOMetaPostPoem(this.articles[1]) : null
    }
  }
})
