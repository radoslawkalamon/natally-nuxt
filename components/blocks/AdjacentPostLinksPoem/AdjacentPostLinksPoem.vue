<template>
  <div class="adjacent-post-links-poem">
    <ComponentsAdjacentPostLinkPrevious
      v-if="previousPoem"
      class="adjacent-post-links-poem__previous"
      label="Poprzednia"
      :path="previousPoem.path"
      :title="previousPoem.title"
    />
    <ComponentsAdjacentPostLinkNext
      v-if="nextPoem"
      class="adjacent-post-links-poem__next"
      label="NastÄ™pna"
      :path="nextPoem.path"
      :title="nextPoem.title"
    />
  </div>
</template>

<script src="./AdjacentPostLinksPoem.ts" lang="ts"></script>
<style src="./AdjacentPostLinksPoem.css" scoped></style>
