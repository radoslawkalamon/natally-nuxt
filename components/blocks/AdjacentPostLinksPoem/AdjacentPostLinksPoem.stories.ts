import { Meta, Story } from '@storybook/vue'
import AdjacentPostLinksPoem from './AdjacentPostLinksPoem.vue'

const meta: Meta = {
  title: 'Blocks / Adjacent Post Links Poem',
  component: AdjacentPostLinksPoem
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: { AdjacentPostLinksPoem },
  props: Object.keys(argTypes),
  template: '<AdjacentPostLinksPoem v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  slug: 'test-post'
}
