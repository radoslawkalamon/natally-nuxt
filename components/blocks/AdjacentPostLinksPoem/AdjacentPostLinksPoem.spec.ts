import { merge } from 'lodash'
import { mount } from '@vue/test-utils'
import AdjacentPostLinksPoem from './AdjacentPostLinksPoem.vue'
import { MockNuxtContent } from '@/devtools/jest.mock.nuxt.content'

jest.mock('@/utils/dto.meta.post.poem', () => ({
  DTOMetaPostPoem: class MockEmptyClass {
    path: string;
    title: string;

    constructor () {
      this.path = '/poezja-314/test-poem'
      this.title = 'Test poem'
    }
  }
}))

const defaultOptionsFactory = (options?: object) => merge({
  mocks: {
    $fetchState: {
      pending: false
    },
    $content: MockNuxtContent(Array(2).fill({}))
  },
  propsData: {
    slug: 'test-slug'
  },
  stubs: [
    'ComponentsAdjacentPostLinkPlaceholder',
    'ComponentsAdjacentPostLinkPrevious',
    'ComponentsAdjacentPostLinkNext'
  ]
}, options)

describe('Blocks / Adjacent Post Links Poem', () => {
  test('shall render', async (): Promise<void> => {
    const wrapper = mount(AdjacentPostLinksPoem, defaultOptionsFactory())
    await (AdjacentPostLinksPoem as any).options.fetch.call(wrapper.vm)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('shall show placeholders when loading', async (): Promise<void> => {
    const wrapper = mount(AdjacentPostLinksPoem, defaultOptionsFactory({
      mocks: {
        $fetchState: {
          pending: true
        }
      }
    }))
    await (AdjacentPostLinksPoem as any).options.fetch.call(wrapper.vm)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
