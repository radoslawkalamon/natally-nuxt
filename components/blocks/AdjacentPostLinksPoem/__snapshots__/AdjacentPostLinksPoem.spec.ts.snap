// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / Adjacent Post Links Poem shall render 1`] = `
"<div data-test=\\"blocks-adjacent-post-links-poem\\" class=\\"adjacent-post-links-poem\\">
  <componentsadjacentpostlinkprevious-stub label=\\"Poprzednia\\" path=\\"/poezja-314/test-poem\\" title=\\"Test poem\\" class=\\"adjacent-post-links-poem__previous\\"></componentsadjacentpostlinkprevious-stub>
  <componentsadjacentpostlinknext-stub label=\\"NastÄ™pna\\" path=\\"/poezja-314/test-poem\\" title=\\"Test poem\\" class=\\"adjacent-post-links-poem__next\\"></componentsadjacentpostlinknext-stub>
</div>"
`;

exports[`Blocks / Adjacent Post Links Poem shall show placeholders when loading 1`] = `
"<div data-test=\\"blocks-adjacent-post-links-poem\\" class=\\"adjacent-post-links-poem\\">
  <componentsadjacentpostlinkplaceholder-stub></componentsadjacentpostlinkplaceholder-stub>
  <componentsadjacentpostlinkplaceholder-stub></componentsadjacentpostlinkplaceholder-stub>
</div>"
`;
